pomegranate/NaiveBayes.pyx

```diff
@@ -138,8 +150,14 @@ cdef class NaiveBayes( object ):
         None
         """
 
-        X = numpy.array(X)
-        y = numpy.array(y)
+        X = _convert(X)
+        y = _convert(y)
+
+        if not isinstance( self.models[0], HiddenMarkovModel ):
+            if X.ndim > 2:
+                raise ValueError("input data has too many dimensions")
+            elif X.ndim == 2 and self.d != X.shape[1]:
+                raise ValueError("input data rows do not match model dimension")
```

Comment:
For 1d input, should check to make sure that the data matches the model dimension too, e.g. `elif X.ndim == 1 and self.d > 1` or something